make/Images.gmk

```diff
@@ -98,16 +109,60 @@ $(eval $(call SetupExecute, jlink_jdk, \
     WARN := Creating jdk image, \
     DEPS := $(JDK_JMODS) $(BASE_RELEASE_FILE) \
         $(call DependOnVariable, JDK_MODULES_LIST, $(JDK_IMAGE_SUPPORT_DIR)/_jlink_jdk.vardeps), \
-    OUTPUT_DIR := $(JDK_IMAGE_DIR), \
+    OUTPUT_DIR := $(JDK_LINK_OUTPUT_DIR), \
     SUPPORT_DIR := $(JDK_IMAGE_SUPPORT_DIR), \
-    PRE_COMMAND := $(RM) -r $(JDK_IMAGE_DIR), \
+    PRE_COMMAND := $(RM) -r $(JDK_LINK_OUTPUT_DIR), \
     COMMAND := $(JLINK_TOOL) --add-modules $(JDK_MODULES_LIST) \
-        $(JLINK_JDK_EXTRA_OPTS) --output $(JDK_IMAGE_DIR) \
+        $(JLINK_JDK_EXTRA_OPTS) --output $(JDK_LINK_OUTPUT_DIR) \
         $(JLINK_DISABLE_WARNINGS), \
 ))
 
 JLINK_JDK_TARGETS := $(jlink_jdk)
 
+ifeq ($(JLINK_PRODUCE_RUNTIME_LINK_JDK), true)
+
+  RL_BUILD_MODULE_NAME := jdk.unsupported_jlink_runtime
+  RL_CREATE_PLUGIN_MOD_OUTPUT := $(SUPPORT_OUTPUTDIR)/$(RL_BUILD_MODULE_NAME)
+  JDK_RUN_TIME_IMAGE_SUPPORT_DIR := $(SUPPORT_OUTPUTDIR)/images/runtime-link-support
+
+  JLINK_RUNTIME_CREATE_ARG := -J--module-path=$(RL_CREATE_PLUGIN_MOD_OUTPUT)
+  JLINK_RUNTIME_CREATE_ARG += -J--add-exports=java.base/jdk.internal.module=$(RL_BUILD_MODULE_NAME)
+  JLINK_RUNTIME_CREATE_ARG += -J--add-exports=java.base/jdk.internal.jimage=$(RL_BUILD_MODULE_NAME)
+  JLINK_RUNTIME_CREATE_ARG += -J--add-exports=jdk.jlink/jdk.tools.jlink.internal=$(RL_BUILD_MODULE_NAME)
+  JLINK_RUNTIME_CREATE_ARG += --create-linkable-runtime jimage=$(JDK_LINK_OUTPUT_DIR)/lib/modules:module-path=$(IMAGES_OUTPUTDIR)/jmods
```

Comment:
I would suggest using recommendation 17 from the [style guideline](https://openjdk.org/groups/build/doc/code-conventions.html) here.
```suggestion
  JLINK_RUNTIME_CREATE_ARG := \
        -J--module-path=$(RL_CREATE_PLUGIN_MOD_OUTPUT) \
        -J--add-exports=java.base/jdk.internal.module=$(RL_BUILD_MODULE_NAME) \
        -J--add-exports=java.base/jdk.internal.jimage=$(RL_BUILD_MODULE_NAME) \
        -J--add-exports=jdk.jlink/jdk.tools.jlink.internal=$(RL_BUILD_MODULE_NAME) \
        --create-linkable-runtime jimage=$(JDK_LINK_OUTPUT_DIR)/lib/modules:module-path=$(IMAGES_OUTPUTDIR)/jmods \
        #
```
or just inline as it's only used in one location.

Reply:
No longer applicable.